netbox/utilities/querysets.py

```diff
@@ -39,6 +39,12 @@ def restrict(self, user, action='view'):
                     # Any permission with null constraints grants access to _all_ instances
                     attrs = Q()
                     break
+            else:
+                # for else, when no break
+                # avoid duplicates when JOIN on many-to-many fields without using DISTINCT.
+                # DISTINCT acts globally on the entire request, which may not be desirable.
+                allowed_objects = self.model.objects.filter(attrs)
+                attrs = Q(pk__in=allowed_objects)
```

Comment:
This approach can be optimized by retrieving only the IDs list (rather than a full queryset) in the first query.

```suggestion
                allowed_object_ids = self.model.objects.filter(attrs).values_list('pk', flat=True)
                attrs = Q(pk__in=allowed_object_ids)
```